src/databricks/labs/ucx/azure/credentials.py

```diff
@@ -288,6 +298,14 @@ def _create_storage_credentials_for_storage_accounts(self) -> list[StorageCreden
                 validation_results = self._storage_credential_manager.validate(storage_credential_info, url)
             except BadRequest:
                 logger.warning(f"Could not validate storage credential {storage_credential_info.name} for url {url}")
+            except (PermissionDenied, NotFound):
+                logger.error(
+                    "Error validating all storage credentials. Please check the error message and tr again."
+                    "Rolling back access connectors and storage credentials created."
+                )
+                self._resource_permissions.delete_storage_credential(execution_results)
+                delete_access_connectors = [access_connector for access_connector, _ in access_connectors]
```

Comment:
should be the urls right?
```suggestion
                access_connector_urls = [url for _, url in access_connectors]
```

Reply:
the url is for the storage account which could be like abfss://container1@storageaccount.dfs.windows.net
access_connector would be AccessConnector data class
which will contain values like /subscriptions/subscriptionID/resourceGroups/RGName/providers/Microsoft.Databricks/accessConnectors/connection_name